src/prayer/AnswerPrayerForm/AnswerPrayerFormConnected.js

```diff
@@ -19,33 +38,71 @@ class AnswerPrayerFormConnected extends React.Component {
-                      id: values.id,
-                      answer: values.answer,
-                    },
-                    refetchQueries: () => [
-                      { query: GET_PRAYER_FEED, variables: { type: 'USER' } },
-                    ],
-                  });
-                  this.props.navigation.pop();
-                }}
-                avatarSource={photo}
-                {...this.props}
-                onClose={() => this.props.navigation.pop()}
-                title={"Celebrate God's faithfulness"}
-                prayerId={this.props.navigation.getParam('prayerId', '')}
-                prayerText={this.props.navigation.getParam('prayerText', '')}
-                prayerAnswer={this.props.navigation.getParam(
-                  'prayerAnswer',
-                  ''
+          <Mutation mutation={INTERACT_WITH_PRAYER}>
+            {(interactWithPrayer) => (
+              <Mutation mutation={ANSWER_PRAYER}>
+                {(answerPrayer) => (
+                  <AnswerPrayerForm
+                    loading={profileLoading}
+                    onSubmit={(values) => {
+                      answerPrayer({
+                        variables: {
+                          id: values.id,
+                          answer: values.answer,
+                        },
+                        refetchQueries: () => [
+                          {
+                            query: GET_PRAYER_FEED,
+                            variables: { type: 'USER' },
+                          },
+                        ],
+                      });
+                      this.props.navigation.pop();
+                    }}
+                    avatarSource={photo}
+                    {...this.props}
+                    onClose={() => this.closePrayer()}
+                    title={"Celebrate God's faithfulness"}
+                    prayerId={this.prayerId}
+                    prayerText={this.props.navigation.getParam(
+                      'prayerText',
+                      ''
+                    )}
+                    prayerAnswer={this.prayerAnswer}
+                    action={
+                      this.prayerAnswer ? (
+                        <OptionMenu>
+                          <ActionComponent
+                            component={
+                              <OptionMenuText>Remove answer</OptionMenuText>
+                            }
+                            options={[
+                              {
+                                title: 'Remove Answer',
+                                method: async () => {
+                                  await interactWithPrayer({
+                                    variables: {
+                                      id: this.prayerId,
+                                      action: 'REMOVE_ANSWER',
+                                    },
+                                  });
+                                  await this.closePrayer();
```

Comment:
don't need to await this

Reply:
I just removed the `closePrayer` function in favor of using `this.props.navigation.pop()`. And yeah, removed the `await`. ;)